editor.planx.uk/src/@planx/components/Pay/Public/Pay.test.tsx

```diff
@@ -21,21 +22,92 @@ jest
 const resumeButtonText = "Resume an application you have already started";
 const saveButtonText = "Save and return to this application later";
 
-it("renders correctly (is hidden) with <= £0 fee", () => {
-  const handleSubmit = jest.fn();
+const flowWithUndefinedFee: Store.flow = {
+  _root: {
+    edges: ["setValue", "pay"],
+  },
+  setValue: {
+    type: TYPES.SetValue,
+    edges: ["pay"],
+    data: {
+      fn: "application.fee.payable",
+      val: "0",
+    },
+  },
+  pay: {
+    type: TYPES.Pay,
+    data: {
+      fn: "application.fee.typo",
+    },
+  },
+};
 
-  // if no props.fn, then fee defaults to 0
-  setup(<Pay handleSubmit={handleSubmit} />);
+const flowWithZeroFee: Store.flow = {
+  _root: {
+    edges: ["setValue", "pay"],
+  },
+  setValue: {
+    type: TYPES.SetValue,
+    edges: ["pay"],
+    data: {
+      fn: "application.fee.payable",
+      val: "0",
+    },
+  },
+  pay: {
+    type: TYPES.Pay,
+    data: {
+      fn: "application.fee.payable",
+    },
+  },
+};
 
-  // handleSubmit is still called to set auto = true so Pay isn't seen in card sequence
-  expect(handleSubmit).toHaveBeenCalled();
-});
+// Mimic having passed setValue to reach Pay
+const breadcrumbs: Breadcrumbs = {
+  setValue: {
+    auto: true,
+    data: {
+      "application.fee.payable": ["0"],
+    },
+  },
+};
+
+describe("Pay component when fee is undefined or £0", () => {
+  beforeEach(() => {
+    getState().resetPreview();
+  });
+
+  it("Shows an error if fee is undefined", () => {
+    const handleSubmit = jest.fn();
+
+    setState({ flow: flowWithUndefinedFee, breadcrumbs: breadcrumbs });
+    expect(getState().computePassport()).toEqual({
+      data: { "application.fee.payable": ["0"] },
+    });
+
+    setup(<Pay fn="application.fee.typo" handleSubmit={handleSubmit} />);
+
+    // handleSubmit has NOT been called (not skipped), Pay shows error instead
+    expect(handleSubmit).not.toHaveBeenCalled();
+    expect(
+      screen.getByText("We are unable to calculate your fee right now"),
+    ).toBeInTheDocument();
+    expect(screen.queryByText("Continue")).not.toBeInTheDocument();
+  });
+
+  it("Skips pay if fee = 0", () => {
+    const handleSubmit = jest.fn();
+
+    setState({ flow: flowWithZeroFee, breadcrumbs: breadcrumbs });
+    expect(getState().computePassport()).toEqual({
+      data: { "application.fee.payable": ["0"] },
+    });
 
-it("should not have any accessibility violations", async () => {
-  const handleSubmit = jest.fn();
-  const { container } = setup(<Pay handleSubmit={handleSubmit} />);
-  const results = await axe(container);
-  expect(results).toHaveNoViolations();
```

Comment:
This accessibility level test at the top-level of Pay was actually meaningless (it was testing the "skipped scenario" :ghost:), so now there's 3 accessibility tests running against the `Confirm` child component container: 
![Screenshot from 2023-10-06 09-44-45](https://github.com/theopensystemslab/planx-new/assets/5132349/01861daf-6d1e-48c4-9142-26009b334d91)